File: Programmers/week4/42583/kwonjin2/42583.js
function solution(bridge_length, weight, truck_weights) {
  let time = 0;
  let currentWeight = 0;
  let queue = Array.from({ length: bridge_length }).fill(0);

  while (truck_weights.length > 0 || currentWeight > 0) {
    time++;
    currentWeight -= queue.shift();

    if (currentWeight + truck_weights[0] <= weight) {
      const newTruck = truck_weights.shift();
      queue.push(newTruck);
      currentWeight += newTruck;
    } else {
      queue.push(0);
    }
  }

  return time;
}

File: Programmers/week4/42583/jjunii/42583.py
from collections import deque

def solution(bridge_length, weight, truck_weights):
    queue = deque([0]*bridge_length)    # bridge
    total_weight, answer = 0, 0         # total_weight on bridge
    trucks = deque(truck_weights)
    
    while trucks:
        out_weight = queue.popleft()
        total_weight -= out_weight
        answer += 1
        
        if total_weight + trucks[0] <= weight:
            in_weight = trucks.popleft()
            queue.append(in_weight)
            total_weight += in_weight
        else:
            queue.append(0)

    return answer + bridge_length